### PR TITLE
clean(backend): remove searches and getAlls that were used by explorer or not even used at all

### PR DESCRIPTION
**Pull request by shawnyama**
# Description

* Deleting anything search-like that was used in the data explorer
* Deleting unused "getAll" endpoints

### CODE REVIEW

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java [hunk 1]
```diff
@@ -378,65 +339,6 @@ ResponseEntity<Model> getModelFromConfigId(
...
-				)
-			),
-			@ApiResponse(
-				responseCode = "500",
-				description = "There was an issue retrieving models from the data store",
-				content = @Content
-			)
-		}
-	)
-	public ResponseEntity<List<Model>> searchModels(
```
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java:400]: @kbirk do we want to strike this or will it still be useful for the new project-page searching?

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/proxies/climatedata/ClimateDataProxy.java [hunk 2]
```diff
@@ -12,9 +12,6 @@ public interface ClimateDataProxy {
 	@GetMapping("/status/{uuid}")
 	ResponseEntity<JsonNode> status(@PathVariable("uuid") String uuid);
 
-	@GetMapping("/search/esgf")
```
Comment (dvince2) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/proxies/climatedata/ClimateDataProxy.java:15]: Removing the searching of climate data effectively drops support for climate data as users won't have a way to get this data into terrarium. This is more a problem for @dgauldie to be aware of!

**Review by dvince2 — APPROVED** [review]
Looks good! Just wait on @kbirk to confirm the model search code before merging?

**Review by mwdchang — APPROVED** [review]

**Review by YohannParis — APPROVED** [review]
Have you made sure that no services uses those endpoints by checking @dgauldie dashboard?

**Comment by shawnyama** [comment]
> Have you made sure that no services uses those endpoints by checking @dgauldie dashboard?

I shall do this next week

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java (hunk 1 above)
Comment (kbirk) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelController.java:399]: Nah that's good to nuke. The guts of it is in the service.

**Review by kbirk — APPROVED** [review]

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/search/SearchByAssetTypeController.java
Comment (kbirk) [inline packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/search/SearchByAssetTypeController.java:1]: Actually, can you keep this file? We will piggy back on this code for the project search

File: packages/client/hmi-client/src/services/search.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import API from '@/api/api';
+import { AssetType, TerariumAssetThatSupportsAdditionalProperties } from '@/types/Types';
+import { logger } from '@/utils/logger';
+
+export async function searchByAssetType(
+	text: string,
+	assetType: AssetType
+): Promise<TerariumAssetThatSupportsAdditionalProperties[]> {
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/search.ts:8]: This should return the same type than used for `AssetType`.

**Review by YohannParis — APPROVED** [review]

File: packages/server/src/main/java/software/uncharted/terarium/hmiserver/proxies/climatedata/ClimateDataProxy.java (hunk 2 above)
Comment (shawnyama) [reply packages/server/src/main/java/software/uncharted/terarium/hmiserver/proxies/climatedata/ClimateDataProxy.java:15]: Yeah, you know what I'll add this search back in and we can worry about this if we want to clean up climate data. I can see that happening.

File: packages/client/hmi-client/src/services/search.ts (hunk 3 above)
Comment (shawnyama) [reply packages/client/hmi-client/src/services/search.ts]: I tried some ways of doing that using generics but it doesn't really work. AssetType is an enum.

**Pull request merged**